awx/main/dispatch/worker/callback.py

```diff
@@ -26,7 +30,37 @@ class CallbackBrokerWorker(BaseWorker):
 
     MAX_RETRIES = 2
 
+    def __init__(self):
+        self.buff = {}
+        self.last_write = tz_now()
+
+    def read(self, queue):
+        try:
+            return queue.get(block=True, timeout=.25)
+        except QueueEmpty:
+            return 'FLUSH'
+
+    def flush(self):
+        should_flush = (
```

Comment:
The main optimization is to avoid an insert *per* event, but instead buffer event saves using `bulk_create`.  The way this works is that we'll buffer *up to* 2000 events at once.

Additionally, in the work loop that pulls events over the IPC pipe, we time out every quarter second, and use a special "FLUSH" message to signal that we should flush.

So effectively, inserts will buffer for *up to* a quarter second.  If there's a lot coming in, we're doing bulk inserts to cut down on individual insert + commit costs, and this is making a *very* notable performance different for high-write workloads.